builtin/credential/aws/path_login.go

```diff
@@ -1364,7 +1364,7 @@ func (b *backend) pathLoginUpdateIam(ctx context.Context, req *logical.Request,
 	if roleEntry.InferredEntityType == ec2EntityType {
 		instance, err := b.validateInstance(ctx, req.Storage, entity.SessionInfo, roleEntry.InferredAWSRegion, callerID.Account)
 		if err != nil {
-			return logical.ErrorResponse(fmt.Sprintf("failed to verify %s as a valid EC2 instance in region %s", entity.SessionInfo, roleEntry.InferredAWSRegion)), nil
+			return logical.ErrorResponse("failed to verify %s as a valid EC2 instance in region %s: %s", entity.SessionInfo, roleEntry.InferredAWSRegion, err), nil
```

Comment:
nit:
```suggestion
			return logical.ErrorResponse("failed to verify %s as a valid EC2 instance in region %s: %w", entity.SessionInfo, roleEntry.InferredAWSRegion, err), nil
```

Reply:
Unfortunately this won't work here. The format string is interpreted with `fmt.Sprintf`, not `fmt.Errorf`: https://github.com/hashicorp/vault/blob/master/sdk/logical/response.go#L116

Reply:
Nvm, this this on `logical.ErrorResponse`...